### ISSUE TITLE
Issue with data import

### ISSUE THREAD

**Issue by alex83803** [nghia-vo/sarepy#3]
Dear Sarepy team,

first of all, thank you so much for the great support so far.

I have been working with own sinogram pictures and applied some basic filters to it. I came to the same conclusion as you. When I apply a median filter to the whole picture, the whole picture is blurred and it is definitely better to come up with an artifact filtering solution like yours. 

I wanted to test your code on some of my sinogram files (1D arrays I assume). As far as I have understood, your code requires 2D arrays. If I try to run your code as is with a 1D sinogram file, I get value errors.  (ValueError: not enough values to unpack (expected 2, got 1))

Then I had a look at your Dataset in order to see which shape your data have (https://doi.org/10.5281/zenodo.1443568). 

In the folders I found some .tif data and one .hdf data when I print the shape of the data, I get a shape corresponding to a 1D array. 
For instance if I take the rec__orig.tif data (Data_Fig23_Fig24 folder) and look at its shape I get:
The first 10 values of Array are: [18761    42     8     0    10   256     4     1     0  2560]
The shape of the array is: 13107267

Which corresponds to a 1D array.

Here is the code I ran to check the shape of the Array: 
import numpy as np
image = open("/Documents/Python/data/rec__orig.tif", "r")
mat0 = np.fromfile(image, dtype=np.uint16, count=-1, sep='') 
values = mat0[:10]
print('The first 10 values of Array are:', values)
(shape, ) = mat0.shape
print('The shape of the array is:', shape)

So I came to the idea of tweaking the array dimension issue by making a 2D array out the initial 1D array.
To do so, I wrote the following code: 
fileorig = open("/Documents/Python/source/rec__orig.tif", "r")
sinogram1= np.fromfile(fileorig, dtype=np.uint16, count=-1, sep='') 
sinogram2 = np.asarray([sinogram1, ])
sinogram3 = remove_all_stripe(sinogram2, 3, 81, 31)
sinogram4 = np.array(sinogram3, dtype=np.uint16)
sinogram4.tofile('/Documents/Python/cleaned.bin')

When I add this piece of code at the bottom of your stripe_artifact_removal_original.py file and run it, I get the following error: 
stripe_artifact_removal.py:239: RuntimeWarning: invalid value encountered in true_divide
  listfact = listdiff / listdiffbck

The only piece of your code I managed to run without error is the remove_stripe_based_sorting function.
fileorig = open("/Documents/Python/source/rec__orig.tif", "r")
sinogram1= np.fromfile(fileorig, dtype=np.uint16, count=-1, sep='') 
sinogram2 = np.asarray([sinogram1, ])
sinogram3 = remove_stripe_based_sorting(sinogram2, 31)
sinogram4 = np.array(sinogram3, dtype=np.uint16)
sinogram4.tofile('/Documents/Python/cleaned.bin')

I am convinced that I do something wrong while importing the sinogram file (np.fromfile parameters). 
Maybe i define the wrong dtype or use a wrong function to import the data. 

I would be very grateful if you could have a look at my issue and tell me what I am doing wrong. I guess it is a small detail making a big difference. 

Basically the core question here is: How would you run the remove_all_stripes code on the rec__orig.tif sinogram file?

Thanks a lot for your help and have a great weekend.

**Comment by nghia-vo** [comment]
Hi Alex,
    As I said before, the methods need to be applied to the sinogram (where the ring artifacts appears as stripe artifacts). So you have to apply my codes to "sino_orig.tif" not "rec_orig.tif". To get rec_orig.tif you need to apply some reconstruction methods which available somewhere else (https://www.astra-toolbox.com/docs/algs/index.html#d-cpu-algorithms)
   Tutorial about tomgoraphy: https://www.mcs.anl.gov/research/projects/X-ray-cmt/rivers/tutorial.html
   Back to your issue, it has nothing to do with the algorithms but the python programming skills. You need to improve this before playing around with algorithms. 
   For example this command (shape, ) = mat0.shape just returns the height of the image which is 1D value (Image is a 2D array height x width). The conclusion that my dataset is a 1D array is quite stunning to me. 
   I won't reply further to issues related to Python programming errors, Alex. You have to improve it yourself I'm afraid.

**Comment by alex83803** [comment]
Good afternoon,

First of all thank you for answering my message. 
I have to give you right. My python programming skills definitely need improvement. 
Moreover, tomography is something new to me and I didn’t really know what to start with.
All that said, your support have helped me a lot and I am grateful for that.

I wish you a good day.

Regards

Alex
